app/datasources/db/models.py

```diff
@@ -1,9 +1,64 @@
-from typing import Optional
+from sqlmodel import JSON, Column, Field, SQLModel, UniqueConstraint, select
 
-from sqlmodel import Field, SQLModel
 
+class SqlQueryBase:
 
-class Contract(SQLModel, table=True):
-    address: bytes = Field(nullable=False, primary_key=True)
+    @classmethod
+    async def get_all(cls, session):
+        result = await session.exec(select(cls))
+        return result.all()
+
+    async def _save(self, session):
+        session.add(self)
+        await session.commit()
+        return self
+
+    async def update(self, session):
+        return await self._save(session)
+
+    async def create(self, session):
+        return await self._save(session)
+
+
+class AbiSource(SqlQueryBase, SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    name: str = Field(nullable=False)
+    url: str = Field(nullable=False)
+
+
+class Abi(SqlQueryBase, SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    abi_hash: bytes = Field(nullable=False, index=True, unique=True)
+    relevance: int | None = Field(nullable=False, default=0)
+    abi_json: dict = Field(default_factory=dict, sa_column=Column(JSON))
+    source_id: int | None = Field(
+        nullable=True, default=None, foreign_key="abisource.id"
+    )
+
```

Comment:
Maybe a `Relationship` should be added? https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/#declare-relationship-attributes

source: AbiSource | None = Relationship(back_populates="abis")

Reply:
I didn´t add them because I would like to try them better. I could read that are not really fields on database, I´d say that we could go with them for now and take a look where are necessary. https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/#what-are-these-relationship-attributes

Reply:
We can try how it works.